dom.go

```diff
@@ -139,8 +139,37 @@ func (h *HTML) restoreText(prev *HTML) {
 	}
 }
 
-func (h *HTML) restoreHTML(prev *HTML) {
-	h.node = prev.node
+func (h *HTML) mutate(prev *HTML) {
+	// Check for compatible tag and mutate previous instance on match, otherwise start fresh
```

Comment:
The following block is basically the block from old line 257 of Restore in this diff.